lib/sisimai/bite/email/domino.rb

```diff
@@ -0,0 +1,174 @@
+
+        require 'sisimai/address'
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        subjecttxt = ''     # (String) The value of Subject:
+        v = nil
+
+        hasdivided.each do |e|
+          next if e.empty?
+
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
+              break if blanklines > 1
+              next
+            end
+            rfc822list << e
+
+          else
+            # Before "message/rfc822"
+            next if readcursor & Indicators[:deliverystatus] == 0
+
+            # Your message
+            #
+            #   Subject: Test Bounce
+            #
+            # was not delivered to:
+            #
+            #   kijitora@example.net
+            #
+            # because:
+            #
+            #   User some.name (kijitora@example.net) not listed in Domino Directory
+            #
+            v = dscontents[-1]
+
+            if e =~ /\Awas not delivered to:\z/
+              # was not delivered to:
+              if v['recipient']
+                # There are multiple recipient addresses in the message body.
+                dscontents << Sisimai::Bite.DELIVERYSTATUS
+                v = dscontents[-1]
+              end
+              v['recipient'] ||= e
+              recipients += 1
+
+            elsif cv = e.match(/\A[ ][ ]([^ ]+[@][^ ]+)\z/)
+              # Continued from the line "was not delivered to:"
+              #   kijitora@example.net
+              v['recipient'] = Sisimai::Address.s3s4(cv[1])
+
+            elsif e =~ /\Abecause:\z/
+              # because:
+              v['diagnosis'] = e
+
+            else
+
+              if v['diagnosis'] && v['diagnosis'] == 'because:'
+                # Error message, continued from the line "because:"
+                v['diagnosis'] = e
+
+              elsif cv = e.match(/\A[ ][ ]Subject: (.+)\z/)
+                #   Subject: Nyaa
+                subjecttxt = cv[1]
+              end
+            end
+          end
+        end
+
```

Comment:
__[rubocop]__
* Style/IfInsideElse: Convert `if` nested inside `else` to `elsif`.

<!-- Comment by [SideCI](https://sideci.com) -->